### PR TITLE
Apply token to data with circular references

### PR DESCRIPTION
**Pull request by rcjpisani**
Fixes issues with circular references in data provided (https://github.com/rcjpisani/redactyl.js/issues/12)

### CODE REVIEW

**Comment by rcjpisani** [comment]
@MatthewClair @vesse - would you be able to review when you have the chance?

File: README.md [hunk 1]
```diff
@@ -1,41 +1,67 @@
...
+  // Required
+  'properties': [ 'apiKey', 'password', 'phone' ],
+
+  // Optional
+  'text': '[REDACTED]',
+  'replacer': myCustomReplacerFunction
 });
 ```
+
+Optional - provide your own replacer function:
```
Comment (vesse) [inline README.md:32]: I would add a mention or a link to eg. MDN about this. It was obvious looking at the code that it is the replacer for `JSON.stringify` but here I assumed that it's meant for custom way of redacting values.

File: src/index.d.ts [hunk 2]
```diff
@@ -12,7 +16,15 @@ declare namespace Redactyl {
...
     text?: string;
+
+    /**
+     * Custom replacer function
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter|MDN}
+     */
+    replacer?: (key: string, value: RedactData) => RedactData;
   }
+
+  function getReplacer(): (key: string, value: RedactData) => RedactData;
```
Comment (vesse) [inline src/index.d.ts]: This is not really available from the module (not even in plain JS?)

File: README.md [hunk 3]
```diff
@@ -44,5 +70,8 @@ Add the names of properties that should be redacted.
 `setText`(`text`)
 Set the text to replace the redacted properties with. Default: [REDACTED]
 
+`setReplacer`(`function`)
```
Comment (vesse) [inline README.md:75]: Typedef missing for `setReplacer`

File: src/replacer.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+function getReplacer() {
```
Comment (vesse) [inline src/replacer.js:3]: This is not referenced at all in `index.js` nor exported to be manually given in by the user. Seems to work but impossible to use?

**Review by vesse — COMMENTED** [review]
Looks to me this does not yet solve circular structure handling

File: src/replacer.js (hunk 4 above)
Comment (rcjpisani) [reply src/replacer.js:3]: Silly mistake, but good catch! I've decided that it's unlikely that using the default replacer will have an impact on projects using an older version currently. I've gone ahead and made it use the internal replacer module by default, therefore, no need to expose that module for users.

**Pull request merged**